### ISSUE TITLE
Vm not found

### ISSUE THREAD

**Issue by bogi0704** [Telmate/terraform-provider-proxmox#241]
Hi everyone,

perfoming "terraform apply" results in the following error:

`proxmox_vm_qemu.proxmox_vm[0]: Creating...

Error: Vm 'debian-10-template' not found`

My main.tf:

```
terraform {
  required_providers {
    proxmox = {
      source = "local/provider/proxmox"
      version = ">=1.0.0"
    }
  }
}

provider "proxmox" {
    pm_api_url = "https://192.168.178.9:8006/api2/json"
    pm_user = "proxmoxtf@pam"
    pm_password = "****"
    pm_tls_insecure = "true"

  pm_log_enable = true
  pm_log_file = "terraform-plugin-proxmox.log"
  pm_log_levels = {
    _default = "debug"
    _capturelog = ""
  }
}

resource "proxmox_vm_qemu" "proxmox_vm" {
  count                 = 1
  name                  = "tf-vm-${count.index}"
  target_node           = "ProxmoxServer1"


  clone                 = "debian-10-template"
  #clone                        = "9001"
  #clone                        = "VM 9002"
  full_clone            = true

  os_type               = "cloud-init"
    cores               = 2
    sockets             = "1"
    cpu                 = "host"
    memory              = 2048
    scsihw              = "virtio-scsi-pci"
    bootdisk            = "scsi0"

  disk {
      #id                = 0
      size              = "20G"
      type              = "scsi"
      storage           = "lvm-thin"
      #storage_type      = "lvm"
      iothread          = true
  }

  network {
      #id                = 0
      model             = "virtio"
      bridge            = "vmbr0"
  }

# Cloud Init Settings
  ipconfig0 = "ip=192.168.178.15${count.index + 1}/24,gw=192.168.178.1"

  sshkeys = <<EOF
    ${var.ssh_key}
    EOF

}
```

Terraform v0.13.5

Proxmox provider v2.6.2

**Comment by doransmestad** [comment]
@bogi0704   Thanks for reporting this!   Quick question: is the template you are trying to clone from on the same proxmox node (ProxmoxServer1)?

**Comment by bogi0704** [comment]
@doransmestad Hi, thanks for looking at this issue! Yes it is:
![Proxmox](https://user-images.githubusercontent.com/39438845/98483656-743b6900-220a-11eb-868f-842ea33c7c08.PNG)

I have tried to select the template in multiple ways. None of them worked.

**Comment by doransmestad** [comment]
@bogi0704   I'm not able to reproduce the error you're seeing  :disappointed:   The clone is working for me as expected even after copy-pasting your resource block (and adapting minimally to enable it to run on my test host).   Here's the verbatim from my test to prove I didn't change much:

```
resource "proxmox_vm_qemu" "proxmox_vm" {
  count                 = 1
  name                  = "tf-vm-${count.index}"
  target_node           = "testproxmox"

  clone                 = "pxtest-10-test"
  full_clone            = true

  os_type               = "cloud-init"
  cores               = 2
  sockets             = "1"
  cpu                 = "host"
  memory              = 2048
  scsihw              = "virtio-scsi-pci"
  bootdisk            = "scsi0"

  disk {
      #id                = 0
      size              = "20G"
      type              = "scsi"
      storage           = "local-lvm"
      #storage_type      = "lvm"
      iothread          = true
  }

  network {
      #id                = 0
      model             = "virtio"
      bridge            = "vmbr0"
  }

  # Cloud Init Settings
  ipconfig0 = "ip=192.168.178.15${count.index + 1}/24,gw=192.168.178.1"

  sshkeys = <<EOF
    EOF

}
```

I see in your provider block you're using a local copy of the provider?   Did you build from source here?   I presume you still see the same error even if you use the official binary from the registry?  https://registry.terraform.io/providers/Telmate/proxmox/latest

Another thought:  can you change the name of your template and see if that makes a difference?  I can't imagine it would, but I'm running low on ideas here  :sweat_smile:

**Comment by bogi0704** [comment]
@doransmestad Thanks again for looking at this issue!

Yes I have built it from source. I have just tried it again using the official binary from the registry. The error still occurs.
The client that I am using terraform on is a WSL1 Ubuntu VM. I thought this may be an issue so I spun up a debian VM and tried it from there. The issue is still the same. 

I have also changed the name of the template. I just called it "test", but the error is still coming up. 
Is there a way to reference the template by it's ID (for example id=9000) rather than using the name?

I looked through the debug log but could not find an issue. I am a rookie terraform user though...
[Terraform Output.txt](https://github.com/Telmate/terraform-provider-proxmox/files/5512913/Terraform.Output.txt)
